### PR TITLE
DOC: Add zorder example to docs

### LINKED ISSUE #1224: DOC: add explanation of using zorder to control which layers are on top
See eg https://stackoverflow.com/questions/59052898/why-are-line-plots-in-geopandas-ignoring-plot-order-when-it-comes-to-display-lay (@martinfleis was faster in answering ;))

We should add some explanation to the mapping docs on this.

### PR DESCRIPTION
**Pull request by Sangarshanan**
Closes #1224

### CODE REVIEW

**Comment by martinfleis** [comment]
Hi,
sorry for a slow response. I would prefer using our `naturalearth_cities` and `naturalearth_lowres` datasets for this example.

```py
cities = geopandas.read_file(geopandas.datasets.get_path('naturalearth_cities'))
world = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))
```

You don't need to import `matplotlib`, as you can make axis using one plot and reuse it for another. 

```py
ax = cities.plot(color='k')
world.plot(ax=ax)

ax = cities.plot(color='k', zorder=2)
world.plot(ax=ax, zorder=1)
```

I would also make a note than lower zorder = lower layer. Thanks!

**Comment by Sangarshanan** [comment]
Sure...Will do that

File: doc/source/mapping.rst [hunk 1]
```diff
@@ -200,6 +200,25 @@ Before combining maps, however, remember to always ensure they share a common CR
     @savefig capitals_over_countries_2.png
     plt.show();
 
+Control the order of multiple layers in a map
+------------------------------------------------
+
+When plotting multiple layers, use zorder to take control of the order of layers being rendered.
```
Comment (martinfleis) [inline doc/source/mapping.rst]:
```suggestion
When plotting multiple layers, use ``zorder`` to take control of the order of layers being plotted. 
```

File: doc/source/mapping.rst [hunk 2]
```diff
@@ -200,6 +200,25 @@ Before combining maps, however, remember to always ensure they share a common CR
     @savefig capitals_over_countries_2.png
     plt.show();
 
+Control the order of multiple layers in a map
+------------------------------------------------
+
+When plotting multiple layers, use zorder to take control of the order of layers being rendered. 
+Lower the zorder, lower the layer is on the map and vice versa
```
Comment (martinfleis) [inline doc/source/mapping.rst]:
```suggestion
The lower the ``zorder`` is, the lower the layer is on the map and vice versa.
```

File: doc/source/mapping.rst [hunk 3]
```diff
@@ -200,6 +200,25 @@ Before combining maps, however, remember to always ensure they share a common CR
     @savefig capitals_over_countries_2.png
     plt.show();
 
+Control the order of multiple layers in a map
+------------------------------------------------
+
+When plotting multiple layers, use zorder to take control of the order of layers being rendered. 
+Lower the zorder, lower the layer is on the map and vice versa 
+
+Without zorder cities gets plotted below world
```
Comment (martinfleis) [inline doc/source/mapping.rst]:
```suggestion
Without specified ``zorder``, cities (Points) gets plotted below world (Polygons), following the default order based on geometry types.
```

File: doc/source/mapping.rst [hunk 4]
```diff
@@ -200,6 +200,25 @@ Before combining maps, however, remember to always ensure they share a common CR
...
+
+.. ipython:: python
+
+    ax = cities.plot(color='k')
+    world.plot(ax=ax)
+
+We can set the zorder of cities higher than world so that it's on top
```
Comment (martinfleis) [inline doc/source/mapping.rst]:
```suggestion
We can set the ``zorder`` for cities higher than for world to move it of top.
```

**Comment by Sangarshanan** [comment]
Thanks for helping out : )

File: doc/source/mapping.rst [hunk 5]
```diff
@@ -200,6 +200,25 @@ Before combining maps, however, remember to always ensure they share a common CR
     @savefig capitals_over_countries_2.png
     plt.show();
 
+Control the order of multiple layers in a plot
+~~~~~~~~~~~~~~~~
```
Comment (martinfleis) [inline doc/source/mapping.rst]:
```suggestion
----------------------------------------------
```

This is the correct one :).

**Comment by martinfleis** [comment]
Oh, one final thing. You need to tell `sphinx` that images should be saved. And you were right with the header type before. My bad. I can't push changes (as you did not allow it), so pasting the correct version below.

```rst
Control the order of multiple layers in a plot
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When plotting multiple layers, use ``zorder`` to take control of the order of layers being plotted.
The lower the ``zorder`` is, the lower the layer is on the map and vice versa.

Without specified ``zorder``, cities (Points) gets plotted below world (Polygons), following the default order based on geometry types.

.. ipython:: python

    ax = cities.plot(color='k')
    @savefig zorder_default.png
    world.plot(ax=ax);

We can set the ``zorder`` for cities higher than for world to move it of top.

.. ipython:: python

    ax = cities.plot(color='k', zorder=2)
    @savefig zorder_set.png
    world.plot(ax=ax, zorder=1);
```

**Comment by Sangarshanan** [comment]
Oh... I have allowed edits from maintainers

I will make the change tho... Thank you for your patience

**Comment by martinfleis** [comment]
Thank you @Sangarshanan!

**Pull request merged**